### PR TITLE
Add ™ to Airflow in prominent places

### PR DESCRIPTION
**Pull request by potiuk**
We've started the process of registering the trademark of Airflow and we need to add ™ marks in prominent places of our website.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow/index.rst [hunk 1]
```diff
@@ -15,10 +15,10 @@
     specific language governing permissions and limitations
     under the License.
 
-What is Airflow?
+What is Airflow ™?
```
Comment (pankajkoti) [inline docs/apache-airflow/index.rst]:
```suggestion
What is Airflow™?
```

File: docs/apache-airflow/core-concepts/index.rst [hunk 2]
```diff
@@ -18,7 +18,7 @@
 Core Concepts
 =============================
 
-Here you can find detailed documentation about each one of Airflow's core concepts and how to use them, as well as a high-level :doc:`architectural overview <overview>`.
+Here you can find detailed documentation about each one of the core concepts of Apache Airflow™ and how to use them, as well as a high-level :doc:`architectural overview <overview>`.
```
Comment (pankajkoti) [inline docs/apache-airflow/core-concepts/index.rst:21]: Are we registering the trademarks for `Apache Airflow` or just `Airflow`?

Comment (potiuk) [reply docs/apache-airflow/core-concepts/index.rst:21]: Just aiirflow. But you should read it as Apache (Airflow™)

Comment (potiuk) [reply docs/apache-airflow/core-concepts/index.rst:21]: Generally speaking it's a bit of dualism:
1) ASF expects us to use `Apache Airflow` in prominent places where it is mentioned for the first time on page /paragraph or chapter.
2) The trademark people (us patent office) will expect to see installation page and there specifically should be "Airlfow™" without Apache.

The 1)  is "Brand" , the 2) is "Trademark" and those two are slightly different and serve slightly different purpose. 

We will never get it *right* or *clearly defined* IMHO - the Trademark/Brand does not really `compile` in a consistent way I am afraid - my attempt is to make "best effort" to pass the Trademark registration and keep ASF branding guidelines fulfilled.

Not water + fire, but close.

Comment (pankajkoti) [reply docs/apache-airflow/core-concepts/index.rst:21]: okay, understood, thank you!

**Review by hussein-awala — APPROVED** [review]
Looks good!
I'm wondering if we need to update/add something in the pre-commits to force using `Airflow™` instead of `Airflow`

**Review by pierrejeambrun — APPROVED** [review]

**Comment by potiuk** [comment]
> Looks good!
I'm wondering if we need to update/add something in the pre-commits to force using Airflow™ instead of Airflow

I don't think we need to use ™ everywhere, I actually aske the trademarks @ asf to finf out answers to some of the questions I have there. I am quite sure there willl be many places where "Airflow" is still good - and the answer "where to use ™" is not one that will be easy to put in automated script to decide, I am afraid.

**Pull request merged**